Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -2059,6 +2078,48 @@ void MediaPlayerPrivateGStreamer::handleMessage(GstMessage* message)
         playbin3SendSelectStreamsIfAppropriate();
         break;
     }
+    case GST_MESSAGE_STREAM_START: {
+        // Real track id configuration in MSE is managed by AppendPipeline.
+        if (isMediaSource())
+            break;
+
+        auto player = m_player.get();
+        bool updated = false;
+        for (auto& track : m_audioTracks.values()) {
+            String streamId = trackIdFromPadStreamStartOrNull(*track->pad());
```

Comment:
This is not going to work in mediastream case, the track uses a GstStream in this case, not a pad, which is null, which triggers crashes, see the EWS results.

Reply:
Thanks. Yes. I've noticed the crashes and the next version of the patch will include code to bail out in the case of isMediaStream(), as well as in the case that there's no pad attached to the track. Still, even with those two changes, there were still a lot of tests failing, many of them because of redundant trackAdded events caused by the mechanism of removing the track and readding it.
I changed approach and delayed the track creation (only for the non-MSE and non-MediaStream cases) until the stream-start message. That left only 11 regressions which I still have to debug.